### ISSUE TITLE
Document how to use Elastic Search & MongoDB containers from uwazi-docker with non-containerized uwazi

### ISSUE THREAD

**Issue by fititnt** [fititnt/uwazi-docker#12]
The steps to run uwazi from host (e.g. developer machine) with dockerized Elastic Search + MongoDB where deleted on recent commits.

Since the docker plays even nicer for local development, let's document it again, and put here or at the huridocs repository.

**Comment by fititnt** [comment]
![fork-uwazi](https://user-images.githubusercontent.com/812299/39087980-7020a3da-4580-11e8-8fe3-57b1c09e962c.png)

**Comment by fititnt** [comment]
![yarn hot](https://user-images.githubusercontent.com/812299/39088499-a3d0783c-4589-11e8-997a-1703b4013617.png)
![uwazi welcome screen](https://user-images.githubusercontent.com/812299/39088501-a8d3c5aa-4589-11e8-9997-bfbd64820f28.png)

**Comment by fititnt** [comment]
TODO: report that `yarn install` step must be documented

**Comment by fititnt** [comment]
Good catch! At some point when more stable, we can merge to official docs. Even the docker & docker-compose configs are on public domain (or MIT), so even if I get busy and can't anwser fast and there is sufficient people inside Uwazi to maintain updated huridocs not only could, but should copy even without  the need of asking first.

This also means for people who can use this uwazi-docker as base prepare for containerization other great humanitarian FOSS out there.

**Comment by fititnt** [comment]
Also this tweet can make you and other people aware of what I will say

https://twitter.com/fititnt/status/987747799495213060

**Simple reason is this: machine learning systems have much more documentation and tutorials based tech who could have a bad social impact (think _Uberization of jobs_, unethical mass surveillance, etc) than technology who could mitigate social issues caused by the exponential power of future intelligent systems.**

Maybe I'm going too far about how to connect point A to point B, but I think one more more inexpensive way to empower a lot of HFOSS is abstracting part of it's complexity and give some quick starts for non-specialists in the software stack, but good at other tech. For example, a lot of machine learning software are written in Python, not in JavaScript (used in this case by Uwazi Software) so is more harder for expert volunteers help. Or maybe people on universities who could use Uwazi integrated with other systems for some published papers. Or people who could promote AI hackatons in the future and have a few softwares with some work already done, so instead of people writting hackaton code from scratch and never be used again, we could start have some better contributions.

Using Uwazi as example, it's stack also uses Mongo and Elastic Search, and for external people would be nice that uwazi-docker also gives some suggestion on GUI usages, so help on make easier #4 and #5 (thinking in the perspective of non Mongo / Elastic Search previous specialists) would also improve the quick start for possible researchers on the future (or improve chances about Uwazi being selected on AI Hackatons).

Will take some time to me discover other things that could be useful to add at the [docker-compose.yml](https://github.com/fititnt/uwazi-docker/blob/master/docker-compose.yml) (for who does not know, is a file used for people just use a command to run additional software, e.g. that could allow people use software like Uwazi as part of some experiment platform). In general the people who work with artificial intelligence work do their jobs in softwares like [Tensorflow](https://www.tensorflow.org/) / [Theano](https://github.com/Theano/Theano) /  [Keras](https://keras.io/), but there are others less powerful but simpler to use to discover patterns without coding (or, if use code, platforms that are less complicated) and that could be embed on "toolkits" like a uwazi-docker draft.

In this case, Uwazi uses 2 databases, **Mongo** and **Elastic Search**. Some software who do some machine learning using these data storage or (maybe just other softwares who create nice dashboards based on statistics) could be embed and used without need of put "on the cloud" because even one old computer with 4 or 8GB of RAM could do the work locally (if have access to datasets). **Since more HFOSS uses databases like Mongo and Elastic Search, the same knowledge and documentation can serve for multiple projects.**

This repository have a public domain as default license on propose. Uwazi is just one of the software I'm looking on, but I'm doing this aiming smaller open source projects. Some of these projects embed some machine learning tools but do not document very well, and maybe some quickstarts or infrastructure as code to use as inspiration could foster results at scale.

**Comment by txau** [comment]
@fititnt I'm not sure I'm fully following you, but I hope I can provide some useful information.

We are already working in parallel in machine learning implementations for Uwazi. Most of the research we are doing involves, NLToolkit, Fasttext, Word2Vec and Tensorflow. Everything is coded in python in the server side and we integrate it with Uwazi via an http API. 

There is a branch of Uwazi with the prototype for the front-end: https://github.com/huridocs/uwazi/tree/ML. This is just a scaffolding for a future production ready feature, but so far so good. At the moment we are focusing in text classification, which comes very handy for tagging large datasets.

Next steps involve feature extraction and entity reconciliation.

As you mentioned in your twitter thread, the idea is to make this technology available for small organizations with limited resources, so they can achieve more with their data.

**Comment by fititnt** [comment]
The pull request https://github.com/huridocs/uwazi/pull/2559 @vorburger and documentation of https://github.com/huridocs/uwazi#docker already do this.

Is possible to run docker both ways, but https://github.com/huridocs/uwazi is likely have far better documentation to run only the databases on docker while user can edit/contribute to huridocs/uwazi.

This could be documented on fititnt/uwazi-docker, but mostly would means just expose the ports, so not something really complex, so will not do it here.

Closing this issue now
